### PR TITLE
PowerUp_Code

### PR DESCRIPTION
**Pull request by sw180283**
There are some issues were the code won't run if I leave some comments in.

EDIT: Thanks James for helping fix the problem. This is no longer an issue.

### CODE REVIEW

File: Source_Code.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      repaint(canvas)
+      #move image down
+      
+      for px in getPixels(pic):
+        r1=getRed(px)*0.95
+        g1=getGreen(px)*0.95
+        b1=getBlue(px)*0.95
+        newColor=makeColor(r1,g1,b1)
+        setColor(px,newColor)
+        #make image darker
```
Comment (sw180283) [inline Source_Code.txt]: I have to remove this comment for it to work sometimes

**Comment by sw180283** [comment]
I wanted to make my code move the picture up and down in a smooth animation style, however couldn't find a move function so instead used several copyInto functions. How can I make this better?

File: Source_Code.txt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  number=4
+  #Number of times to move up and down
+  
+  for loop in range (0,number):
+  #make image move up then down
+    maxHeight=10
+    #number of pixels move
+    
+    for up in range (0,maxHeight):
+      copyInto(pic, canvas, targetX, targetY-up)
```
Comment (MaddieK19) [inline Source_Code.txt]: If I used an image that wasn't the power up lighting one it would get an error on this line

**Comment by Adrir** [comment]
Were you able to resolve this with your last commit Sam?

**Comment by sw180283** [comment]
No. Perhaps it's just a downside to using JES? Or would using a movie function work better?

File: Source_Code.txt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      setAllPixelsToAColor(canvas, white)
+      copyInto(pic, canvas, targetX, targetY-up)
+      repaint(canvas)
+
+      #make image lighter
+      for px in getPixels(pic):
+        r1=getRed(px)*1.05
+        g1=getGreen(px)*1.05
+        b1=getBlue(px)*1.05
+        newColor=makeColor(r1,g1,b1)
```
Comment (Adrir) [inline Source_Code.txt:35]: Using getColor(getPixel(picture, xPosition, yPosition)) and modifying the R,G,B components in-place may be marginally faster.

File: Source_Code.txt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    maxHeight=10
+
+    #move image up
+    for up in range (0,maxHeight):
+      setAllPixelsToAColor(canvas, white)
+      copyInto(pic, canvas, targetX, targetY-up)
+      repaint(canvas)
+
+      #make image lighter
+      for px in getPixels(pic):
```
Comment (Adrir) [inline Source_Code.txt:31]: As per previous comment, could you move the pixels _and_ modify their color within the same loop?

File: Source_Code.txt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  #make image move up then down
+  for loop in range (0,number):
+    #number of pixels move
+    maxHeight=10
+
+    #move image up
+    for up in range (0,maxHeight):
+      setAllPixelsToAColor(canvas, white)
+      copyInto(pic, canvas, targetX, targetY-up)
```
Comment (Adrir) [inline Source_Code.txt:27]: This seems like a good approach --- however, you could grab the pixel, manipulate it, and place it in the new canvas within the body of a single loop.

**Comment by Adrir** [comment]
Performance is an issue with JES. For this reason, performance is _not_ assessed. That is, you will _not_ be marked down for poor performance. You are, indeed, welcome to use other Python frameworks if you desire to improve performance (e.g. PyGame).

You do have the option of producing a video when you demo your project at the end of the semester. However, I would avoid using the JES video library for the purposes of this project.

**Pull request closed without merging**